core/src/main/java/me/blackvein/quests/listeners/BlockListener.java

```diff
@@ -93,64 +93,69 @@ public void onBlockBreak(final BlockBreakEvent evt) {
                                         }));
                             }
                         }
-                        if (currentStage.containsObjective(placeType)) {
-                            for (final ItemStack is : quester.getQuestData(quest).blocksPlaced) {
-                                if (evt.getBlock().getType().equals(is.getType()) && is.getAmount() > 0) {
-                                    ItemStack toPlace = new ItemStack(is.getType(), 64);
-                                    for (final ItemStack stack : currentStage.getBlocksToPlace()) {
-                                        if (ItemUtil.compareItems(is, stack, true) == 0) {
-                                            toPlace = stack;
+                        // Sorry for nested if
+                        if (!(quest.getOptions().canAllowBreakPlaceCheating())) {
```

Comment:
Not worried about the nesting but please remove the comment. Also there's an unnecessary pair of parenthesis in this if statement.

Reply:
Done